prog/vm/host_nexus.rb

```diff
@@ -245,9 +245,26 @@ def get_boot_id
   end
 
   def available?
-    sshable.cmd("true")
-    true
-  rescue
+    devices = storage_devices.flat_map { |sd| sd.unix_devices.split(',') }
+    ssh_session = sshable.start_fresh_session
+    is_available = false
+
+    begin
+      checks = [
+        -> { check_storage_smartctl(ssh_session, devices) },
+        -> { check_storage_nvme(ssh_session, devices) },
+        -> { check_storage_read_write(ssh_session, devices) },
+        -> { check_storage_kernel_logs(ssh_session, devices) }
+      ]
+      is_available = checks.all?(&:call)
```

Comment:
I did check how long it takes to check all. On many different tries, it takes around ~4.2s on average


I believe my internet connection is not very good and since these are run on top of the ssh connections, my numbers are not very reliable. Because it takes so long, sometimes I get a log from the monitor saying resource is locked but on its retries, it is resolved